app/controllers/application_controller.rb

```diff
@@ -96,4 +95,17 @@ def set_headers
   def invalid_recaptcha_score?
     recaptcha_reply["score"] < SUSPICIOUS_RECAPTCHA_THRESHOLD
   end
+
+  def strip_param_whitespaces
```

Comment:
Does this strip nested params? Also is there a reason why you iterate over the keys like this?

Reply:
- Well caught, I only went down one nesting level, which only gets us down to the top-level form params.
- I iterate over the keys to find any hashes, because any form params will be nested in a hash. But this can be made neater.

Reply:
If you want you can do something recursive, which tends to make heads hurt a little bit but means we account for any level of nesting

Reply:
I had some fun making it recursive. Thanks for the idea.